pkg/asset/installconfig/vsphere/vsphere.go

```diff
@@ -2,10 +2,348 @@
+
+// vCenterClient contains the login info/creds and client for the vCenter.
+// They are contained in a single struct to facilitate client creation
+// serving as validation of the vCenter, username, and password fields.
+type vCenterClient struct {
+	VCenter    string
+	Username   string
+	Password   string
+	Client     *vim25.Client
+	RestClient *rest.Client
+}
+
+// networkNamer declares an interface for the object.Common.Name() function.
+// This is needed because find.NetworkList() returns the interface object.NetworkReference.
+// All of the types that implement object.NetworkReference (OpaqueNetwork,
+// DistributedVirtualPortgroup, & DistributedVirtualSwitch) and perhaps all
+// types in general embed object.Common.
+type networkNamer interface {
+	Name() string
+}
+
 // Platform collects vSphere-specific configuration.
 func Platform() (*vsphere.Platform, error) {
-	return &vsphere.Platform{}, nil
+	vCenter, err := getClients()
+	if err != nil {
+		return nil, err
+	}
+
+	finder := find.NewFinder(vCenter.Client)
+	ctx := context.TODO()
+
+	dc, dcPath, err := getDataCenter(ctx, root, finder, vCenter.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster, err := getCluster(ctx, dcPath, finder, vCenter.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	datastore, err := getDataStore(ctx, dcPath, finder, vCenter.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	network, err := getNetwork(ctx, dcPath, finder, vCenter.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	apiVIP, ingressVIP := getVIPs()
+
+	platform := &vsphere.Platform{
+		Datacenter:       dc,
+		Cluster:          cluster,
+		DefaultDatastore: datastore,
+		Network:          network,
+		VCenter:          vCenter.VCenter,
+		Username:         vCenter.Username,
+		Password:         vCenter.Password,
+		APIVIP:           apiVIP,
+		IngressVIP:       ingressVIP,
+	}
+	return platform, nil
+}
+
+// getClients() surveys the user for username, password, & vcenter.
+// Validation on the three fields is performed by creating a client.
+// If creating the client fails, an error is returned.
+func getClients() (*vCenterClient, error) {
+	var vcenter, username, password string
+	survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "vCenter",
+				Help:    "The hostname of the vCenter to be used for installation.",
+			},
+			Validate: survey.Required,
+		},
+	}, &vcenter)
+
+	survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Username",
+				Help:    "The username to login to the vCenter.",
+			},
+			Validate: survey.Required,
+		},
+	}, &username)
+
+	survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "Password",
+				Help:    "The password to login to the vCenter.",
+			},
+			Validate: survey.Required,
+		},
+	}, &password)
+
+	// Survey does not allow validation of groups of input
+	// so we perform our own validation.
+	logrus.Infof("Connecting to vCenter %s", vcenter)
```

Comment:
i think we should move this to debug.

Reply:
I used `Info` because there is a noticeable delay for the network connection. This lets the user know what's happening and that the process is not frozen. I'm willing to change but wanted to explain the reasoning.